### PR TITLE
feat: gui windows installer + sshnpd service

### PR DESCRIPTION
**Pull request by Xlin123**
**What I did**
 Created a windows installer to allow easy & supported installs on windows machines
 Created a proper windows service to allow sshnpd to interact with the OS like other windows services. 
 Programmatically signs the executables in the windows binary as part of the "NoPorts" app, in order to avoid any issues with anti-viruses.
 
**How I did it**
  I looked into WiX, VS Installer and Inno Setup and all of them are almost impossible to use for our use case... so instead I started to create a WPF App as the installer. The one downside to this is I couldn't get it to compile as a msi... but all the benefits of an msi can be used from the windows C++ dlls' (See ServiceInstaller.cs). 
  
  So with that here are some windows features we now support 🤠 

-  Event Logging in the Event Viewer
-  sshnpd as a service (with recovery)
-  app uninstalls the windows way
-  registry entry for service configs
-  all executables downloaded from our bundle get signed by the installer 
-  service and app descriptions 
- the installer has a self-contained runtime, and is **now** a lot bigger, 169MB..

NEW OF SEPT 22
 - Onboarding
 - Enrollment
 - Approve/Deny
  
** How to verify it**
[ I'll build it once I'm done and throw it in here probably.](https://filebin.net/b1ywaor1nzcsv0vb)
 
**Important Note**
Create a directory in `C:\Program Files\NoPorts\sshnp\` and place the SshnpdService.exe inside that directory.

The SshnpdService.exe should be inside the windows bundle, but I haven't created a release for it yet.
 
**- Description for the changelog**
feat: gui windows installer + sshnpd service

### CODE REVIEW

File: tools/windows-installer/SshnpdService/Properties/PublishProfiles/FolderProfile.pubxml.user [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
...;True|2024-07-26T15:55:46.5701039-04:00||;True|2024-07-26T15:55:18.9211215-04:00||;True|2024-07-19T11:54:46.6638001-04:00||;True|2024-07-19T09:56:58.7676027-04:00||;True|2024-07-19T09:45:59.0839738-04:00||;True|2024-07-19T09:45:44.9391570-04:00||;True|2024-07-19T09:45:15.6295347-04:00||;True|2024-07-19T09:44:53.5987691-04:00||;True|2024-07-19T09:44:43.0197788-04:00||;True|2024-07-19T09:38:38.6530424-04:00||;True|2024-07-16T09:33:58.2810892-04:00||;True|2024-07-16T09:30:02.6686572-04:00||;True|2024-07-15T15:08:49.4516190-04:00||;True|2024-07-15T14:50:27.0512831-04:00||;True|2024-07-15T14:46:09.7622736-04:00||;True|2024-07-15T10:46:03.8522878-04:00||;True|2024-07-15T10:44:14.0120102-04:00||;True|2024-07-15T10:36:19.8709496-04:00||;True|2024-07-15T10:35:50.1838770-04:00||;True|2024-07-15T10:35:24.0095129-04:00||;True|2024-07-15T10:34:10.5379066-04:00||;True|2024-07-15T10:33:30.0626619-04:00||;True|2024-07-15T10:32:25.9836209-04:00||;True|2024-07-15T10:22:43.5004201-04:00||;True|2024-07-15T09:57:14.8760545-04:00||;True|2024-07-12T09:11:11.7089159-04:00||;True|2024-07-11T09:48:51.6520694-04:00||;True|2024-07-10T14:09:27.2054713-04:00||;True|2024-07-08T14:26:25.3664065-04:00||;</History>
```
Comment (xavierchanth) [inline tools/windows-installer/SshnpdService/Properties/PublishProfiles/FolderProfile.pubxml.user]: What is this for? Looks like this is supposed to be gitignored?

**Review by xavierchanth — DISMISSED** [review]
Other than the one concern, lgtm

File: tools/windows-installer/SshnpdService/Properties/PublishProfiles/FolderProfile.pubxml.user (hunk 1 above)
Comment (Xlin123) [reply tools/windows-installer/SshnpdService/Properties/PublishProfiles/FolderProfile.pubxml.user]: resolved in [c79a957](https://github.com/atsign-foundation/noports/pull/1217/commits/c79a957b67e97342cfca7bc8a4d3710adc7771b8)

**Comment by Xlin123** [comment]
There's some features I'd like to add in the future. I'll throw them into tickets once this is merged.

**Review by cconstab — COMMENTED** [review]
We should probably remive the binaries from the github tree

**Review by cpswan — CHANGES_REQUESTED** [review]
@cconstab is correct. The binaries should not be checked in. I'd suggest adding a rule to .gitignore

**Comment by Xlin123** [comment]
@cpswan changes made in [4a18b58](https://github.com/atsign-foundation/noports/pull/1217/commits/4a18b5899861ceb8f167ce51ec1dc25c859d4667)

**Comment by xavierchanth** [comment]
Oops... I don't have commit signing setup on Windows.

**Review by xavierchanth — APPROVED** [review]

**Review by cpswan — APPROVED** [review]

**Pull request merged**